### PR TITLE
fix: manifest delta num may overflow

### PR DESCRIPTION
**Pull request by jiacai2050**
## Rationale
Fix bugs found in local write bench.

## Detailed Changes
- When manifest starts up, the delta num may overflow since it's initialized to 0.
- When manifest merge_update, since the deltas is unsorted, so we may first delete a non-existing file.

## Test Plan
CI

### CODE REVIEW

File: src/metric_engine/src/manifest/encoding.rs [hunk 1]
```diff
@@ -303,18 +303,30 @@ impl Snapshot {
...
+
+        self.records
+            .retain(|record| !to_deletes.contains(&record.id));
         self.header.length = (self.records.len() * SnapshotRecord::LENGTH) as u64;
-        Ok(())
     }
 
     pub fn into_bytes(self) -> Result<Bytes> {
-        let buf = Vec::with_capacity(self.header.length as usize + SnapshotHeader::LENGTH);
+        let buf = Vec::with_capacity(self.header.length as usize * SnapshotHeader::LENGTH);
```
Comment (zealchen) [inline src/metric_engine/src/manifest/encoding.rs:329]: self.header.length as usize + SnapshotHeader::LENGTH is right ?

self.header.length is the total length of the payload of Snapshot in bytes.

**Comment by zealchen** [comment]
"When manifest merge_update, since the deltas is unsorted, so we may first delete a non-existing file."

Why is the failure to delete related to the sorting of the deltas? Intuitively they are mutually irrelevent with each other.

File: src/server/src/main.rs [hunk 2]
```diff
@@ -52,6 +62,16 @@ async fn hello() -> impl Responder {
     HttpResponse::Ok().body("Hello world!")
 }
 
+#[get("/toggle")]
+async fn toggle(data: web::Data<AppState>) -> impl Responder {
```
Comment (zealchen) [inline src/server/src/main.rs:66]: Any reasons for choosing Actix over Axum?

File: src/metric_engine/src/manifest/encoding.rs (hunk 1 above)
Comment (jiacai2050) [reply src/metric_engine/src/manifest/encoding.rs:329]: Good catch!

**Comment by jiacai2050** [comment]
> Why is the failure to delete related to the sorting of the deltas? Intuitively they are mutually irrelevent with each other.

This is a tricky bug. For example, we have tow delta files to merge:
1. to_add: 100
2. to_add:  101, to_delete: 100

If the delta files are unsorted, if we first merge `1`,  100 is not in snapshot yet, so it will delete nothing.

File: src/server/src/main.rs (hunk 2 above)
Comment (jiacai2050) [reply src/server/src/main.rs:65]: No specific reason. GPT says actix is more feature-rich.

**Review by baojinri — APPROVED** [review]
LGTM

**Pull request merged**